content/documentation/pages/5-feature-guides/1-streams/9-function-composition.md

```diff
@@ -1,9 +1,331 @@
 ---
 path: 'feature-guides/streams/function-composition/'
 title: 'Composing Functions'
-description: 'Adding a function to an existing application.'
+description: 'Compose functions to an existing application.'
 ---
 
 With the function composition, you can attach a functional logic dynamically to an existing event streaming application. The business logic is a mere implementation of `java.util.Function`, `java.util.Supplier` or `java.util.Consumer` interfaces that map to `processor`, `source` and `sink`, respectively.
 
 If you have a functional logic implemented using `java.util.Function`, you can represent this `java.util.Function` as a Spring Cloud Data Flow `processor` and attach it to an existing source or sink application. The function composition in this context could be the source and processor combined in one single application: a “new source,” or it could be the processor and sink combined into a single application: “a new sink.” Either way, the transformation logic represented in the `processor` application can be composed into a `source` or `sink` application without having to develop a separate `processor` application.
+
+Let’s consider how a pipeline is created to perform a simple transformation by using three applications and then see how it can be implemented as a pipeline by using two applications that use functional composition.
+
+### **Streaming Pipeline with three applications**
+
+You can setup Spring Cloud Data Flow and Spring Cloud Skipper servers using the installation guide.
+
+For the first stream, we will use the out-of-the-box `http`, `transform` and `log` applications.
+Hence, you make sure to import the out of the box streaming applications into Spring Cloud Data Flow server.
+
+Now we can create a simple stream without function composition:
+
+```
+stream create hello --definition "http --server.port=9000 | transformer --expression=(\"Hello \"+payload.toString().toUpperCase()) | log"
+```
+
+Then we can deploy the stream:
+
+```
+stream deploy hello
+```
+
+```
+http post --data "friend" --target "http://localhost:9000"
+```
+
+You can see the following log message at the `log` application:
+
+```
+[sformer.hello-1] log-sink                                 : Hello FRIEND
+
+```
+
+In this stream, we have the http (source), transformer (processor), and log (sink) applications deployed as standalone applications in the target platform (in this case, it is `local`). For certain use-cases, for a simple payload transformation logic, we might want to have the `Processor` application combined with either the `Source` or `Sink` applications. For instance, transformation scenarios like masking some specific user specific fields at the Source output data doesn’t necessarily need to be deployed as a separate standalone application. Instead, it can be composed either at the Source or Sink application.
```

Comment:
> In this stream, we have the http (source), transformer (processor), and log (sink) applications deployed as standalone applications in the target platform (in this case, it is `local`). For certain use-cases, a simple payload transformation logic may not require a standalone application. It might be beneficial to combine the transformation with either the source or the sink.